### PR TITLE
[13.0] Add connector_importer_source_sftp

### PR DESCRIPTION
**Pull request by simahawk**
Includes #76 

Depends on
- [x] https://github.com/OCA/storage/pull/68
- [x] https://github.com/OCA/connector/pull/387

TODO

- [x] Fix server.env integration

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1 +1,3 @@
 odoo_test_helper
+odoo13-addon-storage_backend @ git+https://github.com/OCA/storage@refs/pull/68/head#subdirectory=setup/storage_backend
+odoo13-addon-storage_backend_sftp @ git+https://github.com/OCA/storage@refs/pull/68/head#subdirectory=setup/storage_backend_sftp
```
Comment (LoisRForgeFlow) [inline test-requirements.txt]: This is already merged.

**Review by LoisRForgeFlow — COMMENTED** [review]
I think that `connector_importer` lacks a small configuration section in the readme.

Also, could you check travis?

File: connector_importer_source_sftp/tests/common.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            {
+                "name": "test_sftp_source",
+                "csv_delimiter": ",",
+                "storage_id": cls.storage_backend.id,
+                "sftp_path_input": "/input",
+                "sftp_path_error": "/error",
+                "sftp_path_success": "/success",
+            }
+        )
+        source._onchance_csv_file()
```
Comment (mmequignon) [inline connector_importer_source_sftp/tests/common.py]: still this onchance :rofl:

**Review by mmequignon — APPROVED** [review]

File: connector_importer_source_sftp/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "summary": """Add import source capable of loading files from SFTP.""",
+    "version": "13.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "Camptocamp,ACSONE,Odoo Community Association (OCA)",
+    "maintainers": ["simahawk", "sebalix"],
+    "website": "https://github.com/OCA/connector-interfaces",
+    "depends": [
+        "connector_importer",
+        "component_event",
+        "server_environment",
```
Comment (sebalix) [inline connector_importer_source_sftp/__manifest__.py]: Is it OK to enforce the use of this dependency to manage the configuration? Not better to use a new module for that?

File: connector_importer_source_sftp/tests/common.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.addons.component.tests.common import (
+    SavepointComponentCase,
+    SavepointComponentRegistryCase,
+)
+from odoo.addons.connector_importer.tests.common import TestImporterMixin
+
+
+class TestSourceCSVSFTPMixin(object):
```
Comment (sebalix) [inline connector_importer_source_sftp/tests/common.py]:
```suggestion
class TestSourceCSVSFTPMixin():
```

File: connector_importer_source_sftp/tests/test_event_listener.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        super().setUp()
+        # Not sure why but this write gets lost if done in setUpClass :/
+        # Probably depends on server env override?
+        # self.source.write(
+        #     {
+        #         "sftp_path_input": "input",
+        #         "sftp_path_error": "error",
+        #         "sftp_path_success": "success",
+        #     }
+        # )
```
Comment (sebalix) [inline connector_importer_source_sftp/tests/test_event_listener.py]: `setupClass` or `setUp`? Is it still relevant? As it is commented and I can't see any piece of code updating these fields for the test?

**Review by sebalix — COMMENTED** [review]
Minor remark + question, LG

File: connector_importer_source_sftp/tests/test_event_listener.py (hunk 5 above)
Comment (simahawk) [reply connector_importer_source_sftp/tests/test_event_listener.py]: cleaning

File: connector_importer_source_sftp/tests/common.py (hunk 4 above)
Comment (simahawk) [reply connector_importer_source_sftp/tests/common.py]: but I like it so much.... :stuck_out_tongue_closed_eyes:

File: connector_importer_source_sftp/__manifest__.py (hunk 3 above)
Comment (simahawk) [reply connector_importer_source_sftp/__manifest__.py]: no, you are right, I trash it

**Comment by simahawk** [comment]
@sebalix all comments attended

**Review by sebalix — APPROVED** [review]

**Pull request merged**